monkey/monkey_island/cc/ui/src/components/utils/ServerUtils.tsx

```diff
@@ -54,6 +61,20 @@ export function getMachineHostname(machine): string {
   return hostname;
 }
 
+export function getMachineFromIP(ip, machines) {
+  let machineFromIP = null;
+
+  for (let machine of machines) {
+    let machineIPs = machine['network_interfaces'].map(network_interface => network_interface.split('/')[0]);
```

Comment:
We're duplicating This logic all over the place. See also https://github.com/guardicore/monkey/pull/2553. Let's make his reusable.

Reply:
Done in #2553.